pkg/controller/master/migration/vmim_controller.go

```diff
@@ -39,16 +43,24 @@ func (h *Handler) OnVmimChanged(_ string, vmim *kubevirtv1.VirtualMachineInstanc
 	}
 
 	abortRequested := isAbortRequest(vmim)
-
+	phase := vmim.Status.Phase
 	// debug log shows, after vmim was deleted, the OnChange here may be called two times
-	logrus.Debugf("syncing vmim %s/%s/%s phase %v abortRequested %v deleted %t", vmim.Namespace, vmim.Name, vmi.Name, vmim.Status.Phase, abortRequested, vmim.DeletionTimestamp != nil)
-	if vmim.Status.Phase == kubevirtv1.MigrationPending && !abortRequested {
+	logrus.Debugf("syncing vmim %s/%s/%s phase %v abortRequested %v deleted %t", vmim.Namespace, vmim.Name, vmi.Name, phase, abortRequested, vmim.DeletionTimestamp != nil)
+
+	if phase == kubevirtv1.MigrationPhaseUnset && !abortRequested {
+		// set StatePending for UI status showing and menu rendering
+		return vmim, h.setVmiMigrationUIDAnnotation(vmi, string(vmim.UID), StatePending)
+	} else if phase == kubevirtv1.MigrationPending && !abortRequested {
+		err := h.setVmiMigrationUIDAnnotation(vmi, string(vmim.UID), StatePending)
+		if err != nil {
+			return nil, err
+		}
 		return vmim, h.scaleResourceQuota(vmi)
-	} else if vmim.Status.Phase != kubevirtv1.MigrationFailed && abortRequested {
+	} else if phase != kubevirtv1.MigrationFailed && abortRequested {
 		return vmim, h.setVmiMigrationUIDAnnotation(vmi, string(vmim.UID), StateAbortingMigration)
-	} else if vmim.Status.Phase == kubevirtv1.MigrationScheduling {
+	} else if phase == kubevirtv1.MigrationScheduling || phase == kubevirtv1.MigrationScheduled || phase == kubevirtv1.MigrationPreparingTarget || phase == kubevirtv1.MigrationTargetReady || phase == kubevirtv1.MigrationRunning {
```

Comment:
I think using a function to replace this long statement would be better for reading like this https://github.com/kubevirt/kubevirt/blob/54ebc9c4580e67714789e590b7e6a048344d7452/staging/src/kubevirt.io/api/core/v1/types.go#L810-L820.

Reply:
The whole function needs a refactor, with the bump on kubevirt v160 and supporting migration target node directly.

Let's optimize then.